source/backend/api/Services/LeaseService.cs

```diff
@@ -548,8 +542,11 @@ private PimsLease AssociatePropertyLeases(PimsLease lease, IEnumerable<UserOverr
             {
                 PimsProperty property = propertyLease.Property;
                 var existingPropertyLeases = _propertyLeaseRepository.GetAllByPropertyId(property.PropertyId);
+                var propertyWithAssociations = _propertyRepository.GetAllAssociationsById(property.PropertyId);
                 var isPropertyOnOtherLease = existingPropertyLeases.Any(p => p.LeaseId != lease.Internal_Id);
                 var isPropertyOnThisLease = existingPropertyLeases.Any(p => p.LeaseId == lease.Internal_Id);
+                var isDisposedOrRetired = existingPropertyLeases.Any(p => p.Property.IsRetired.HasValue)
+                    || propertyWithAssociations?.PimsDispositionFileProperties?.Any(d => d.DispositionFile.DispositionFileStatusTypeCode == DispositionFileStatusTypes.COMPLETE.ToString()) == true;
```

Comment:
question: the old logic had this `x.Property.IsRetired.HasValue && x.Property.IsRetired.Value` to account for NULLABLE retired flag to have a value (and that the value is **true**).

However the new/fixed logic is only checking that "IsRetired" is not null. Shouldn't we also check that it is true as well below?
`existingPropertyLeases.Any(p => p.Property.IsRetired.HasValue)`